Makefile

```diff
@@ -66,10 +66,87 @@ build-all:
 aro: check-release generate
 	go build -ldflags "-X github.com/Azure/ARO-RP/pkg/util/version.GitCommit=$(VERSION)" ./cmd/aro
 
+# Target to deploy the CosmosDB database and container
+.PHONY: deploy-database
+deploy-database:
+	@echo "Deploying CosmosDB database and container..."
+	@az deployment group create \
+		-g "$(RESOURCEGROUP)" \
+		-n "databases-development-$(USER)" \
+		--template-file pkg/deploy/assets/databases-development.json \
+		--parameters \
+			databaseAccountName="$(DATABASE_ACCOUNT_NAME)" \
+			databaseName="$(DATABASE_NAME)" \
+		1>/dev/null || echo "Deployment failed or already exists"
+
+# Target to login to ACR
+.PHONY: acr-login
+acr-login:
+	@echo "Logging into ACR..."
+	@echo "PULL_SECRET: $${PULL_SECRET}"
+	@acr_auth=$$(echo "$${PULL_SECRET}" | jq -r '.auths["arointsvc.azurecr.io"].auth' | base64 -d); \
+	echo "Decoded ACR Auth: $$acr_auth"; \
+	username=$$(echo "$$acr_auth" | cut -d: -f1); \
+	password=$$(echo "$$acr_auth" | cut -d: -f2); \
+	echo "ACR Username: $$username"; \
+	echo "ACR Password: $$password"; \
+	docker login arointsvc.azurecr.io --username $$username --password $$password
+
+# Target to run the local RP
 .PHONY: runlocal-rp
-runlocal-rp:
-	go run -ldflags "-X github.com/Azure/ARO-RP/pkg/util/version.GitCommit=$(VERSION)" ./cmd/aro rp
-
+runlocal-rp: ci-rp deploy-database acr-login 
+	@set -a; source secrets/env; set +a; \
+	if [ ! -f $(PWD)/aks.kubeconfig ]; then echo "aks.kubeconfig not found"; exit 1; fi; \
+	podman run --rm -p 8443:8443 --network host \
```

Comment:
I get a message locally that:

```
Port mappings have been discarded as one of the Host, Container, Pod, and None network modes are in use
```

Since we use `--network host`, the 8443 port mapping is already implicit. For the sake of clarity, it may be best to not have the `-p` mapping (in any case, it is ignored when using `--network host`)

Reply:
We're discussing this in a Slack thread, but I think to end up with a solution that works on both Mac and Linux, we should remove `--network host` and keep the explicit port mapping. IIUC the "host" in this case is the Podman VM on Macs, and so what ends up happening is that you can talk to the RP via `localhost:8443` from inside the VM but not from the Mac OS.

See this GitHub issue comment for context on `--network host` on Mac OS: https://github.com/containers/podman/issues/15664#issuecomment-1239118559